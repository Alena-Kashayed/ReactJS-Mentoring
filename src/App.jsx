import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';

import Search from './Search/Search';
import Details from './Details/Details';
import MovieList from './MovieList/MovieList';
import Footer from './Footer/Footer';
import styles from './App.scss';
import InfoBar from './InfoBar/InfoBar';
import SortBy from './SortBy/SortBy';
import NotFound from './NotFound/NotFound';
import data from './assets/data.json';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      assets: data,
      // assets: [], // No films found
    };
  }

  handleShowDetails = (movieDetails) => {
    this.setState({
      movieDetails,
    });
  };

  render() {
    const { assets, movieDetails } = this.state;
    const SearchRoute = ({ component: Component, ...rest }) => (
      <Route
        {...rest}
        render={props => (
          <Component
            moviesCount={assets.length}
            {...props}
          />)}
      />
    );
    const MovieDetailsRoute = ({ component: Component, ...rest }) => (
      <Route
        {...rest}
        render={props => (
          <Component
            movieDetails={movieDetails}
            {...props}
          />)}
      />
    );

    return (
      <div className={styles.app}>
        <div className={styles.headerBlock}>
          <Switch>
            <SearchRoute exact path="/" component={Search} />
            <SearchRoute path="/search" component={Search} />
            <MovieDetailsRoute path="/film/:title" component={Details} />
            <Route
              path="*"
              render={() =>
                <NotFound />}
            />
          </Switch>
          {assets.length ? (
            <div className={styles.info}>
              <InfoBar
                movieDetails={movieDetails}
                moviesCount={assets.length}
              />
              <SortBy
                movieDetails={movieDetails}
              />
            </div>) : <div className={styles.info}>{null}</div>
          }
        </div>
        <div className={styles.contentBlock}>
          <Route
            exact
            path="/"
            render={() =>
              <div className={styles.emptyPage}>No films found</div>
            }
          />
          <Route
            path="/search"
            render={() => (<MovieList
              assets={assets}
              handleShowDetails={this.handleShowDetails}
            />)}
          />
        </div>
        <div className={styles.footerBlock}>
          <Footer />
        </div>
      </div>
    );
  }
}

export default App;
