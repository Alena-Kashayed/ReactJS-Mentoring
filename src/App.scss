.app {
  display: flex;
  height: 100%;
  flex-direction: column;

  .headerBlock {
    flex: 0 0 auto;
  }

  .contentBlock {
    flex: 1 0 auto;
    display: flex;
    justify-content: center;
  }

  .footerBlock {
    flex: 0 0 auto;
  }

  .emptyPage {
    color: #C8C8C8;
    display: flex;
    justify-content: center;
    font-size: 60px;
    margin-top: 100px;
  }

  .info {
    background-color: #F5F5F5;
    height: 55px;
    padding: 18px 110px;
    box-sizing: border-box;
    font-weight: 600;
    display: flex;
    justify-content: space-between;
  }
}
