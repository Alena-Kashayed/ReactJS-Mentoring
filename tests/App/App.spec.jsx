import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../../src/App';

describe('App', () => {
  it('is rendered correctly', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
