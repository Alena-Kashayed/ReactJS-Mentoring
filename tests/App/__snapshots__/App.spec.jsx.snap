// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is rendered correctly 1`] = `
<div
  className="app"
>
  <Connect(Loader) />
  <div
    className="headerBlock"
  >
    <Switch>
      <Route
        component={[Function]}
        path="/film/:id"
      />
      <Route
        component={[Function]}
        path="/"
      />
    </Switch>
    <div
      className="info"
    >
      <Connect(InfoBar) />
      <Connect(SortBy) />
    </div>
  </div>
  <div
    className="contentBlock"
  >
    <withRouter(Connect(MovieList)) />
  </div>
  <div
    className="footerBlock"
  >
    <Footer />
  </div>
</div>
`;
